onnxruntime/core/providers/dml/DmlExecutionProvider/src/ExecutionProvider.cpp

```diff
@@ -526,14 +526,30 @@
         std::string partitionKernelPrefix = std::to_string(m_partitionKernelPrefixVal++) + "_";
         uint32_t deviceDataTypeMask = GetSupportedDeviceDataTypeMask();
 
-        return PartitionGraph(
-            graph,
-            *m_internalRegInfoMap,
-            registries,
-            deviceDataTypeMask,
-            m_kernelRegistry.get(),
-            partitionKernelPrefix
-        );
+        bool isFirstPartitionCall = std::none_of(graph.Nodes().begin(), graph.Nodes().end(), [](const onnxruntime::Node& node) {
+                return !node.GetExecutionProviderType().empty() && node.GetExecutionProviderType() == onnxruntime::kDmlExecutionProvider;
+            });
+
+        if (isFirstPartitionCall)
+        {
+            return LightWeightPartitionGraph(
+                graph,
+                *m_internalRegInfoMap,
+                registries,
+                deviceDataTypeMask
+            );
+        } 
+        else
+        {
+            return PartitionGraph(
+                graph,
+                *m_internalRegInfoMap,
+                registries,
+                deviceDataTypeMask,
```

Comment:
>deviceDataTypeMask [](http://example.com/codeflow?start=16&length=18)

Do we still need this? Maybe we do because transformers could have substituted primitive operators (already validated to support that data type) with new aggregate operators (e.g. layernorm) and need another data type check.

Reply:
Thank you, Dwayne for pointing this out. If ORT transformer changes the data type of the new aggregated operator, then we do need to call [IsNodeSupportedByDml()](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphPartitioner.cpp#L248) method in 2nd call of partitioning also. Doing this will actually increase the overall time complexity. 
I think we need to get confirmation from ORT team (someone who is expert on ORT optimizers) on the contract of optimizers i.e. Can optimizers change the data type of the newly aggregated operator or not?